src/osd/ReplicatedPG.h

```diff
@@ -970,11 +971,12 @@ class ReplicatedPG : public PG, public PGBackend::Listener {
   void handle_watch_timeout(WatchRef watch);
 protected:
 
-  ObjectContextRef create_object_context(const object_info_t& oi, SnapSetContext *ssc);
+  ObjectContextRef create_object_context(const object_info_t& oi, SnapSetContext *ssc, bool nocache = false);
   ObjectContextRef get_object_context(
     const hobject_t& soid,
     bool can_create,
-    map<string, bufferlist> *attrs = 0
+    map<string, bufferlist> *attrs = 0,
+    bool nocache = false
```

Comment:
I would be inclined to put nocache after can_create, and have no default value, so the cache/nocache is explicit in every caller.  And make true -> cache, false -> don't cache, as @tchaikov suggested!arg is e